### PR TITLE
【Hackathon 8th No.24】MatterSim 论文复现

### PR DESCRIPTION
**Pull request by BeingGod**
### PR types

New features

### PR changes

Others

### Describe

MatterSim 论文复现

#### 精度对齐

**finetune**

loss（对齐至 1e-4）
loss
MAE of energy（对齐至 3e-4）
MAE_of_energy

原始日志：
paddle：[train.paddle.log](https://github.com/user-attachments/files/20431417/train.paddle.log)
torch：[train.torch.log](https://github.com/user-attachments/files/20431416/train.torch.log)

**inference**
Energy 完全对齐
Forces of first atom 对齐至 1e-7
Stress 对齐至 2e-7

paddle log:
image

torch log:
image

#### 已知问题

1. 不支持分布式微调训练（依赖 paddle_geometric 中分布式 dataloader）
2. ~~不支持stress和strain的梯度反向传播（依赖主框架对 repeat_interleave_grad op 的支持）~~

### CODE REVIEW

**Comment by HydrogenSulfate** [comment]
> 不支持stress和strain的梯度反向传播（依赖主框架对 repeat_interleave_grad op 的支持）

是需要支持 repeat_interleave 的二阶微分吗？ @BeingGod

**Comment by BeingGod** [comment]
> > 不支持stress和strain的梯度反向传播（依赖主框架对 repeat_interleave_grad op 的支持）
> 
> 是需要支持 repeat_interleave 的二阶微分吗？ @BeingGod

是的

File: jointContribution/mattersim/README.md [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+# 5MB
+python src/mattersim/training/finetune_mattersim.py --load_model_path pretrained_models/mattersim-v1.0.0-5M.pdparams --train_data_path tests/data/high_level_water.xyz
+```
+
+## **Known Issues**
+
+* Not support distributed finetune.
+* Not support gradient calculation of stress and strains when training.
+
+# Below is MatterSim's original README
```
Comment (zhiminzhang0830) [inline jointContribution/mattersim/README.md]: original readme可以删除了，可以把其中Reference等部分放到上面来

File: jointContribution/mattersim/src/mattersim/forcefield/m3gnet/modules/edge_encoding.py [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+        else:
+            self.register_buffer(name="bessel_weights", tensor=bessel_weights)
+
+    def forward(self, x: paddle.Tensor) -> paddle.Tensor:
+        """
+        Evaluate Bessel Basis for input x.
+
+        Parameters
+        ----------
+        x : torch.Tensor
```
Comment (zhiminzhang0830) [inline jointContribution/mattersim/src/mattersim/forcefield/m3gnet/modules/edge_encoding.py]: paddle.Tensor，可以全局搜索替换一下

File: jointContribution/mattersim/src/mattersim/utils/convert_pretrained_model_utils.py [hunk 3]
```diff
@@ -0,0 +1,140 @@
+from pathlib import Path
+
+import paddle
+import torch
+
+ROOT_DIR = Path(__file__).resolve().parent.parent.parent.parent
+
+FC_NAMES = [
```
Comment (zhiminzhang0830) [inline jointContribution/mattersim/src/mattersim/utils/convert_pretrained_model_utils.py:8]: 1M和5M中的FC_NAMES好像是不一样的，请确认这样设置对两个预训练模型都适用

File: jointContribution/mattersim/src/mattersim/utils/mlflow_utils.py
Comment (zhiminzhang0830) [inline jointContribution/mattersim/src/mattersim/utils/mlflow_utils.py:1]: 这个文件好像没有用到吧?

File: jointContribution/mattersim/tests/applications/test_phonon.py
Comment (zhiminzhang0830) [inline jointContribution/mattersim/tests/applications/test_phonon.py:1]: 没有用到的文件可以删除

**Review by zhiminzhang0830 — COMMENTED** [review]
paddle_geometric 上传到单独仓库，在readme安装部分添加从对应仓库下载paddle_geometirc的说明命令

**Comment by zhiminzhang0830** [comment]
使用pre-commit格式化一下代码，另外，烦请检查代码，如代码文件未曾使用，可以删除

File: jointContribution/mattersim/tests/applications/test_phonon.py
Comment (BeingGod) [reply jointContribution/mattersim/tests/applications/test_phonon.py:1]: ACK. Thx~

File: jointContribution/mattersim/src/mattersim/utils/mlflow_utils.py
Comment (BeingGod) [reply jointContribution/mattersim/src/mattersim/utils/mlflow_utils.py:1]: ACK. Thx~

File: jointContribution/mattersim/src/mattersim/forcefield/m3gnet/modules/edge_encoding.py (hunk 2 above)
Comment (BeingGod) [reply jointContribution/mattersim/src/mattersim/forcefield/m3gnet/modules/edge_encoding.py]: ACK. Thx~

File: jointContribution/mattersim/README.md (hunk 1 above)
Comment (BeingGod) [reply jointContribution/mattersim/README.md]: ACK. Thx~

**Comment by BeingGod** [comment]
> 使用pre-commit格式化一下代码，另外，烦请检查代码，如代码文件未曾使用，可以删除

File: jointContribution/mattersim/src/mattersim/utils/convert_pretrained_model_utils.py (hunk 3 above)
Comment (BeingGod) [reply jointContribution/mattersim/src/mattersim/utils/convert_pretrained_model_utils.py:8]: double check 了一下，该设置对两个预训练模型都适用

**Comment by BeingGod** [comment]
> 使用pre-commit格式化一下代码，另外，烦请检查代码，如代码文件未曾使用，可以删除

Done.

**Comment by BeingGod** [comment]
finetune command
```bash
python src/mattersim/training/finetune_mattersim.py --load_model_path pretrained_models/mattersim-v1.0.0-5M.pdparams --train_data_path tests/data/high_level_water.xyz
```
Loss
loss

MAE of energy
mae_of_e

MAE of force
mae_of_f

original log:
[train.paddle.log](https://github.com/user-attachments/files/20604126/train.paddle.log)
[train.torch.log](https://github.com/user-attachments/files/20604127/train.torch.log)

**Comment by BeingGod** [comment]
This PR is ready to merge.

File: jointContribution/mattersim/pretrained_models/mattersim-v1.0.0-1M.pdparams
Comment (HydrogenSulfate) [inline jointContribution/mattersim/pretrained_models/mattersim-v1.0.0-1M.pdparams:1]: 权重文件从PR中删掉

File: jointContribution/mattersim/pretrained_models/mattersim-v1.0.0-5M.pdparams
Comment (HydrogenSulfate) [inline jointContribution/mattersim/pretrained_models/mattersim-v1.0.0-5M.pdparams:1]: 权重文件从PR中删掉

File: jointContribution/mattersim/pretrained_models/mattersim-v1.0.0-1M.pdparams
Comment (BeingGod) [reply jointContribution/mattersim/pretrained_models/mattersim-v1.0.0-1M.pdparams:1]: ACK. Thx~

File: jointContribution/mattersim/pretrained_models/mattersim-v1.0.0-5M.pdparams
Comment (BeingGod) [reply jointContribution/mattersim/pretrained_models/mattersim-v1.0.0-5M.pdparams:1]: ACK. Thx~

**Comment by zhiminzhang0830** [comment]
LGTM

**Review by zhiminzhang0830 — APPROVED** [review]

File: jointContribution/mattersim/.gitignore
Comment (HydrogenSulfate) [inline jointContribution/mattersim/.gitignore:1]: test/data下的文件是否可以帮忙上传到bce？

**Review by HydrogenSulfate — APPROVED** [review]
LGTM
